iina/JavasctiptDevTool.swift

```diff
@@ -0,0 +1,506 @@
+          .frame(width: 10, height: 10)
+        Text(title).mono()
+      }
+      .onTapGesture {
+        isExpanded.toggle()
+      }
+      let dataToDisplay = viewType == .array ? Array(data.first(count: 100)) : data
+      if isExpanded {
+        ForEach(dataToDisplay, id: \.0) { item in
+          HStack {
+            Text("\(item.0):").mono().foregroundColor(.secondary)
+              .padding(.leading, 8)
+            Text(item.1).mono()
+              .lineLimit(1)
+          }
+        }
+        if viewType == .array && data.count > 100 {
+          Text("... (\(data.count - 100)) more").mono().foregroundColor(.secondary)
+        }
+      }
+    }
+  }
+}
+
+@available(macOS 12.0, *)
+fileprivate struct ResultView: View {
+  var command: JSEvent
+
+  var body: some View {
+    switch (command.result) {
+    case .nothing:
+      Text("")
+
+    case .string(let value):
+      Text(value).mono().foregroundColor(.brown)
+
+    case .number(let value):
+      Text("\(value)").mono().foregroundColor(.blue)
+
+    case .boolean(let value):
+      Text(value ? "true" : "false").mono().foregroundColor(.purple)
+
+    case .null:
+      Text("null").mono().foregroundColor(.secondary)
+
+    case .undefined:
+      Text("undefined").mono().foregroundColor(.secondary)
+
+    case .array(let title, let data):
+      ObjectView(viewType: .array, title: title, data: data)
+
+    case .object(let title, let data):
+      ObjectView(viewType: .array, title: title, data: data)
+
+    case .opaqueObject(let value):
+      Text(value).mono()
+
+    case .exception(let message, let stack):
+      ZStack {
+        VStack(alignment: .leading) {
+          Text("Exception: \(message)")
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .foregroundColor(.red)
+          Text(stack)
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .foregroundColor(.secondary)
+        }
+        .textSelection(.enabled)
+        .padding(.all, 8)
+      }
+      .background(Color.red.opacity(0.1))
+
+    case .log(let message, let level):
+      ZStack {
+        Text(message).mono()
+          .frame(maxWidth: .infinity, alignment: .leading)
+          .textSelection(.enabled)
+          .padding(.all, 8)
+      }
+      .background(
+        ([.error: .red, .debug: .gray, .verbose: .gray, .warning: .orange] as [Logger.Level: Color])[level]!.opacity(0.1))
+    }
+  }
+}
+
+
+// MARK: - Utils
+
+fileprivate extension Array {
+  func first(count len: Int) -> ArraySlice<Self.Element> {
+    let cnt = len > self.count ? self.count : len
+    return self[0..<cnt]
+  }
+}
+
+@available(macOS 12.0, *)
+struct MonospacedText: ViewModifier {
+  func body(content: Content) -> some View {
+    content
+      .font(.system(.body, design: .monospaced))
+  }
+}
+
+@available(macOS 12.0, *)
+extension View {
+  func mono() -> some View {
+    modifier(MonospacedText())
+  }
+}
```

Comment:
I think I would prefer this to just be a static font property or something, this is a lot of ceremony for something that's really just one line of code